upf/upf_cli.c

```diff
@@ -635,6 +641,82 @@ VLIB_CLI_COMMAND (upf_show_spec_release_command, static) =
 };
 /* *INDENT-ON* */
 
+static clib_error_t *
+upf_node_id_command_fn (vlib_main_t * vm,
+			unformat_input_t * main_input,
+			vlib_cli_command_t * cmd)
+{
+  upf_main_t *gtm = &upf_main;
+  unformat_input_t _line_input, *line_input = &_line_input;
+  clib_error_t *error = NULL;
+  ip46_address_t ip46;
+  u8 *fqdn = 0;
+  pfcp_node_id_t *node_id = &gtm->node_id;
+
+  if (!unformat_user (main_input, unformat_line_input, line_input))
+    return 0;
+
+  while (unformat_check_input (line_input) != UNFORMAT_END_OF_INPUT)
+    {
+      if (unformat (line_input, "fqdn %_%v%_", &fqdn))
+	{
+	  node_id->type = NID_FQDN;
+	  node_id->fqdn = upf_name_to_labels (fqdn);
```

Comment:
shouldn't we free old `fqdn` vector which is no longer in use?

Reply:
We should, done